packages/strapi-plugin-users-permissions/admin/src/components/Permissions/PermissionRow/SubCategory/index.js

```diff
@@ -0,0 +1,131 @@
+import React, { useCallback, useMemo } from 'react';
+import { without } from 'lodash';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { Flex, Padded, Text, Checkbox } from '@buffetjs/core';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import CheckboxWrapper from '../CheckboxWrapper';
+import BaselineAlignment from '../BaselineAlignment';
+import SubCategoryWrapper from './SubCategoryWrapper';
+import { useUsersPermissions } from '../../../../contexts/UsersPermissionsContext';
+import PolicyWrapper from './PolicyWrapper';
+
+const Border = styled.div`
+  flex: 1;
+  align-self: center;
+  border-top: 1px solid #f6f6f6;
+  padding: 0px 10px;
+`;
+
+const SubCategory = ({ subCategory }) => {
+  const {
+    onSelectedAction,
+    onSelectedPermission,
+    onSelectedSubcategory,
+    pluginName,
+    selectedAction,
+  } = useUsersPermissions();
+
+  const actions = useMemo(() => {
+    return Object.values(subCategory.actions).reduce((acc, curr, index) => {
```

Comment:
Could you write some tests for this? Maybe you could extract a function in a utils folder and use it in your `useMemo` so it is easier to test?